src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -177,4 +186,119 @@ public ExamResource(UserService userService, ExamService examService, ExamAccess
+    }
+
+    /**
+     * Post /courses/:courseId/exams/:examId/students : Add multiple users to the students of the exam so that they can access the exam
+     * The passed list of UserDTOs must include the registration number (the other entries are currently ignored and can be left out)
+     * Note: registration based on other user attributes (e.g. email, name, login) is currently NOT supported
+     *
+     * This method first tries to find the student in the internal Artemis user database (because the user is most probably already using Artemis).
+     * In case the user cannot be found, we additionally search the (TUM) LDAP in case it is configured properly.
+     *
+     * @param courseId     the id of the course
+     * @param examId       the id of the exam
+     * @param students     the list of students (with at least registration number) who should get access to the exam
+     * @return             the list of students who could not be registered for the exam, because they could NOT be found in the Artemis database and could NOT be found in the TUM LDAP
+     */
+    @PostMapping(value = "/courses/{courseId}/exams/{examId}/students")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<StudentDTO>> addStudentToCourse(@PathVariable Long courseId, @PathVariable Long examId, @RequestBody List<StudentDTO> students) {
+        log.debug("REST request to add {} as students to exam : {}", students, examId);
+        var course = courseService.findOne(courseId);
+        var instructorOrAdmin = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, instructorOrAdmin)) {
+            return forbidden();
+        }
+        var exam = examService.findOneWithRegisteredUsers(examId);
+        if (!course.equals(exam.getCourse())) {
+            return forbidden();
+        }
+        List<StudentDTO> notFoundStudents = new ArrayList<>();
+        for (var student : students) {
+            var registrationNumber = student.getRegistrationNumber();
+            // 1) we use the registration number and try to find the student in the Artemis user database
+            Optional<User> user = userService.findUserWithGroupsAndAuthoritiesByRegistrationNumber(registrationNumber);
+            if (user.isPresent()) {
+                exam.addUser(user.get());
+                // we only need to add the user to the course group, if the student is not yet part of it, otherwise the student cannot access the exam (within the course)
+                if (!user.get().getGroups().contains(course.getStudentGroupName())) {
+                    userService.addUserToGroup(user.get(), course.getStudentGroupName());
+                }
+                continue;
+            }
+            // 2) if we cannot find the user, we use the registration number and try to find the student in the (TUM) LDAP
+            user = userService.createUserFromLdap(registrationNumber);
```

Comment:
TODO: make sure the user is actually created in the external authentication provider (e.g. JIRA), otherwise the call below `userService.addUserToGroup(..)` will throw an exception

Reply:
fixed